### PR TITLE
Convert soak gke jobs

### PR DESCRIPTION
**Pull request by xiangpengzhao**
#1230 #1829

Sorry for delay. Just create a small PR first to see if the conversion works.

cc @krzyzacy

### CODE REVIEW

**Comment by krzyzacy** [comment]
@k8s-bot ok to test

File: jobs/ci-kubernetes-soak-gke-gci-test.env [hunk 1]
```diff
@@ -0,0 +1,20 @@
+### soak-env
+JENKINS_SOAK_MODE=y
+JOB_NAME=${JOB_NAME//-test/-deploy}
```
Comment (krzyzacy) [inline jobs/ci-kubernetes-soak-gke-gci-test.env]: delete this line, it's handled with that `--soak-test` flag.

Comment (krzyzacy) [inline jobs/ci-kubernetes-soak-gke-gci-test.env]: Same for below

**Comment by krzyzacy** [comment]
@k8s-bot bazel test this

**Comment by xiangpengzhao** [comment]
@krzyzacy I had a short business trip last several days. Sorry for not addressing the PR in time:-(

Failure fixed. I should check the file history and be much more carefully when converting to avoid such basic mistakes as is mentioned in inline comments...

**Comment by krzyzacy** [comment]
@xiangpengzhao no problem, it also mean I need to write better tests to catch them :-)

/lgtm

**Comment by krzyzacy** [comment]
@k8s-ci-robot I can LGTM a PR until you can tell me a joke

/lgtm

**Pull request merged**